### DISCUSSION TITLE
Custom javascript block

### DISCUSSION THREAD

**Discussion by tanujgirish** [postmanlabs/postman-flows#14] [Ideas]
I'd like to see a custom variable decoder, even a freeform javascript block would be enough. Would like to customize the response of one API request to fit the required body structure for the next API request. One example is each key of my object in the response is dynamic, so static mapping with the variables block won't work.

**Comment by ironmanoyo** [comment]
Custom javascript/codable block can have support for `n` inputs and `m` outputs.

**Comment by samay-v** [comment]
Hey @tanujgirish 
I understand your usecase. The ability to transform data within Flows is something we too are looking at closely. The first step towards that was to understand what will be the basic types of data that Flows will be able to deal with, about which you can read more about [here](https://github.com/postmanlabs/postman-flows/blob/main/docs/data-types.md). 

It would be very helpful if you could share a usecase where you'd use a "Javascript block" along with the javascript code you'd like to use there.  cc @ironmanoyo

**Reply by tanujgirish** [reply]
One example would be dynamically changing dictionary keys. Being able to cast those to variables and transform them into any output format I'd like would be ideal. As @WilliamWCYoung mentioned, the use cases for custom javascript are endless. I think having the ability for full flexibility would account for all use cases.

**Reply by akshay2000** [reply]
This use case is still not addressed completely. We have list manipulation blocks now. Could we also get record manipulation blocks? The use case is very simple: The key in response JSON is dependent on a request parameter. This dynamic variable mapping is not possible right now.

**Reply by samay-v** [reply]
Hey @akshay2000, 
After lists, record manipulation is on our roadmap. you can keep an eye on it [here](https://postman-flows.notion.site/postman-flows/ab62d1f9b64044e39ea26894a836f04a?v=87f5928f27b9493fb35f65ef1727e265)

**Reply by akshay2000** [reply]
I did look at the roadmap before posting. Unfortunately, it is not immediately clear from the roadmap whether record manipulation is in the near future. Which roadmap item are you referring to?

**Edit**: Is there a way I can contribute to the features I want?

**Comment by WilliamWCYoung** [comment]
This is related to this discussion, with an actual example: https://github.com/postmanlabs/postman-flows/discussions/106

The examples are literally limitless though, and I worry that there will always be something blocking for someone using this product.
